File: lib/models/competitor.dart
import 'package:eloit/models/item.dart';
import 'package:equatable/equatable.dart';
import 'package:cloud_firestore/cloud_firestore.dart';

class Competitor extends Equatable {
  final String id;
  final Item item;
  final int eloScore;

  const Competitor(
      {required this.id, required this.item, required this.eloScore});

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'item': item.toMap(),
      'eloScore': eloScore,
    };
  }

  @override
  List<Object> get props {
    return [id, item, eloScore];
  }

  @override
  bool get stringify => true;

  Competitor.fromDocumentSnapshot(DocumentSnapshot doc)
      : id = doc.id,
        item = Item.fromMap(doc.get("item")),
        eloScore = (doc.get("eloScore")).toInt();
}

File: db_help/competitor.py
from item import Item

class Competitor:

    def __init__(self, item, cid, eloScore=1400):
        self._item = item
        self._item.add_category(cid)
        self._eloScore = 1400

    def set_id(self, id):
        self._id = id

    def get_id(self):
        return self._id

    def get_name(self):
        return self._item.get_name()

    def toMap(self):
        return {
            "item": self._item.toMap(),
            "eloScore": self._eloScore
        }
